components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -1099,7 +1099,7 @@ int esp_websocket_client_send_with_opcode(esp_websocket_client_handle_t client,
     int ret = ESP_FAIL;
 
     if (client == NULL || len < 0 ||
-            (opcode != WS_TRANSPORT_OPCODES_CLOSE && (data == NULL || len <= 0))) {
+            (opcode != WS_TRANSPORT_OPCODES_CLOSE && len < 0)) {
```

Comment:
Could you please also add a note to the header file, that for sending a zero payload message we have to set x and y to zero?

Reply:
I wonder if we can remove the entire sub condition? Since `len < 0` is taken care of before, and checking opcode with the same length condition seems redundant to me?